test/hotspot/jtreg/runtime/Thread/TooSmallStackSize.java

```diff
@@ -90,6 +92,25 @@ static String getMinStackAllowed(String testOutput) {
         throw new RuntimeException("test fails");
     }
 
+    static ProcessBuilder createProcessWithOptions(String stackOption, String stackSize) throws Exception {
+        if (testShadowSize == null) {
```

Comment:
There would be far less duplication if you just used:

`testShadowSize == null ? "" : ("-XX:StackShadowPages=" + testShadowSize)`

Reply:
Ok, sure.  I'll see if it compiles.

Reply:
The process builder doesn't like a null argument and thinks it's the class name.  There's 3 lines of duplication.

Reply:
That is a shame.